collection-scripts/common.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+# shellcheck disable=SC2034,SC2086,SC2001
+
+export DST_DIR="must-gather"
+
+# run must-gather in the namespaces one by one
+function run_mustgather() {
+    for ns in "$@"; do
+        oc adm inspect "${log_collection_args}" "namespace/$ns" --all-namespaces --dest-dir "$DST_DIR" || echo "Error inspecting namespace/$ns"
+    done
+}
+
+# get the list of namespaces where defined resources exist
+function get_all_namespace() {
+    local nslist
+    for kind in "$@"; do
+        nslist+=$(oc get "$kind" --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.namespace}{"\n"}{end}')
+    done
+    uniq_list "$nslist"
+}
+
+# remove dupplicated namespaces
+function uniq_list() {
+    echo "$@" | sort -u
+}
+
+# get the version of rhoai operator
+function version() {
+  version=''
+
+  # get version from rhoai csv
+  csv_name=$(oc get csv|grep rhods|awk '{print $1}')
+  [[ z == z"${version}" && z != z"${csv_name}" ]] && version=$(oc get csv "$(oc get csv|grep rhods|awk '{print $1}')" -o json|jq .spec.version |xargs)
+
+  # if version not found, get version from rhods operator pod's container
+  operator_name=$(oc get pod -n redhat-ods-operator --ignore-not-found |grep redhat-ods-operator |awk '{print $1}')
+  [[ z != z"${operator_name}" ]] && version=$(oc exec ${operator_name} -n redhat-ods-operator -c manager -- env|grep OPERATOR_CONDITION_NAME|awk -F'=' '{print $2}'| sed -n -r -e 's/.*([[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+(:?\.[[:digit:]])?(:?-[^@]+)?).*/\1/p')
```

Comment:
Will that ```sed -n -r -e '/OPERATOR_CONDITION_NAME/s/.*=.*([[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+(:?\.[[:digit:]])?(:?-[^@]+)?).*/\1/p'``` work the same as the sequence of grep, awk and sed calls?

Reply:
i changed the check.
the current one requires executing in the pod, new check is based on deployment env settings.

some code in the "common.sh" are copied from old implementation, so i kept them as-was when moving around into new files.